cmd/goal/asset.go

```diff
@@ -175,21 +176,29 @@ func lookupAssetID(cmd *cobra.Command, creator string, client libgoal.Client) {
 			"creator account is unknown.")
 	}
 
-	response, err := client.AccountInformation(creator)
+	response, err := client.AccountInformationV2(creator, true)
 	if err != nil {
 		reportErrorf(errorRequestFail, err)
 	}
 
 	nmatch := 0
-	for id, params := range response.AssetParams {
-		if params.UnitName == assetUnitName {
-			assetID = id
-			nmatch++
+	if response.CreatedAssets != nil {
+		for _, asset := range *response.CreatedAssets {
+			params := asset.Params
+			if params.UnitName == nil && assetUnitName == "" {
```

Comment:
@algochoi For my understanding:  What motivates the change in matching logic here for empty unit name?

I'm not following how it's different than the check below.  If the check is retained, I think it'd be prudent to extend (or create) an `e2e_sub` proving the behavior works as intended.

Reply:
There is an existing test that tests behavior that when a single asset with an empty unit name is created, `goal` needs to be able to look for the asset, given the account and an empty unit name. In the v2 models, the unit name would be nil instead of an empty string, so this check is for that.

https://github.com/algorand/go-algorand/blob/8b212cbe65d62ebfcaf8e6fcc5b7f6cdaef21ccf/test/e2e-go/cli/goal/expect/goalAccountInfoTest.exp#L64

Reply:
@algochoi Resolving - Thanks for the context + pointer to existing test.  The PR makes sense as is.

Like I did here, I still sometimes struggle to disambiguate pointer vs value semantics in Go.